frontend/src/pages/SettingsPage.jsx

```diff
@@ -1,16 +1,34 @@
-import { View, Text, StyleSheet, TouchableOpacity, Image, Dimensions } from "react-native";
+import { 
+  View, 
+  Text, 
+  StyleSheet, 
+  TouchableOpacity, 
+  Modal,
+  TextInput,
+  Button,
+  ScrollView,
+  Image, 
+  Dimensions } from "react-native";
 import SlidingButton from "../components/SlidingButton";
 import React, { useEffect, useState } from "react";
 import LocalHost from '../components/data/LocalHost';
 
-const About = ({ enterConnect }) => {
+const Settings = ({ enterConnect }) => {
   const ipAddress = LocalHost.ipAddress;
 
   const [deviceName, setDeviceName] = useState("");
   const [deviceStatus, setDeviceStatus] = useState("");
   const [buttonText, setButtonText] = useState("Change Device");
   const [buttonDisabled, setButtonDisabled] = useState(false);
   const [dotCount, setDotCount] = useState(0);
+  const [modalVisible, setModalVisible] = useState(false);
+  const [userData, setUserData] = useState({
```

Comment:
Is there a specific reason why the values are `null` instead of `0`?

Reply:
They should all be set to false not zero.

Reply:
Setting the default value to null instead of 0 assigns two potential types to each property, requiring additional checks to see if the value is currently `null` or a `number` type each time they are accessed. This can be useful if such explicit checks are necessary (for example, if certain functions shouldn't run if the value is `null`), but otherwise, setting the default value to 0 seems like the better option. Let me know what you think :)

Reply:
Just to clarify, I'm referring to `userData` properties, not the other states. Sorry if I confused you!